dashboard/src/main/java/org/openedx/dashboard/presentation/DashboardFragment.kt

```diff
@@ -101,6 +110,9 @@ import org.openedx.core.R as CoreR
 class DashboardFragment : Fragment() {
 
     private val viewModel by viewModel<DashboardViewModel>()
+    private val iapViewModel by viewModel<IAPViewModel> {
```

Comment:
I think IAPViewModel should be renamed to interactor(for example), and injected inside another viewModel instead of fragment

Reply:
As all the views for the IAP flow work as an overlay on other views and UI State, there are two implementation options.

- **IAP as ViewModel:** 
     - Will provide the Seaprate and shared IAP UI state logic, and can be used for Overlay UI.
- **IAP as Interactor OR Manager:** 
     - Each ViewModel with the IAPInteractor must manage the IAP UI State Logic separately, which will create extra complexity in case of IAP flow change.
     - Extra complexity for billing purchase listeners each ViewModel needs to set up the purchase listener.

@k1rill thoughts please?

Reply:
My opinion is to use **IAP as Interactor or Manager**, it will be harder to implement this while development, but will be much easier to maintain this. Also overlay as a separate compose function has issues on the UI level, I suppose it's better to use a dialog fragment as a wrapper, with the injection of the **IAP as Interactor or Manager** into it's viewModel

Reply:
This UI issues you can see on GIFs to this PR (blinking screen)

Reply:
> My opinion is to use IAP as Interactor or Manager, it will be harder to implement this while development, but will be much easier to maintain this. Also overlay as a separate compose function has issues on the UI level, I suppose it's better to use a dialog fragment as a wrapper, with the injection of the IAP as Interactor or Manager into it's viewModel

This approach seems more appropriate here to fix the blinking screen issue with the IAP ViewModel.

I'm planning to use the following structure.

Screen(MyCourses | Course Dashboard Container | Settings) <--> IAP Framgent-Dialog(Overlay) <--> IAP-ViewModel <--> IAP-Interactor <--> Billing-Process | IAP-Repositroy.